packages/core/src/routes/session.ts

```diff
@@ -504,27 +504,53 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
   );
 
   router.post(
-    '/session/forgot-password/phone/verify-passcode-and-reset-password',
+    '/session/forgot-password/email/send-passcode',
+    koaGuard({ body: object({ email: string().regex(emailRegEx) }) }),
+    async (ctx, next) => {
+      const { jti } = await provider.interactionDetails(ctx.req, ctx.res);
+      const { email } = ctx.guard.body;
+      ctx.userLog.email = email;
+      ctx.userLog.type = UserLogType.ForgotPasswordEmail;
+
+      assertThat(
+        await hasUserWithEmail(email),
+        new RequestError({ code: 'user.email_not_exists', status: 422 })
+      );
+      const { id } = await findUserByEmail(email);
+      ctx.userLog.userId = id;
+      const { usernameAndPassword } = await findUserSignInMethodsById(id);
+      assertThat(usernameAndPassword, 'user.username_password_signin_not_exists');
```

Comment:
i think this will be cool to proceed? since we can use phone / email to set a password

Reply:
Do you mean even if the username does not exist, we can still use a verified phone number or email to identify the userId? As a result, a username is not necessary to identify whether an account can be signed-in with a password? We can still sign in with a password and use phone/email instead to identify the unique userId.

Reply:
per discussion offline, we'll revert forgot password logic due to logic concerns

Reply:
let's add an issue for that instead

Reply:
Updated, PTAL.

Also, LOG-1847 is added to remove username+password sign-in method availability check for phone forgot password flow and keep error code consistency; LOG-1846 is added to manually revert forgot password flow logics and corresponding codes.